pennylane/io.py

```diff
@@ -122,11 +109,177 @@ def from_qiskit(quantum_circuit, measurements=None):
     Returns:
         function: the PennyLane template created based on the ``QuantumCircuit`` object
 
-    .. details::
-        :title: Usage Details
+    **Example:**
+
+    .. code-block:: python
+
+        import pennylane as qml
+        from qiskit import QuantumCircuit
+
+        qc = QuantumCircuit(2, 2)
+        qc.rx(0.785, 0)
+        qc.ry(1.57, 1)
+
+        my_qfunc = qml.from_qiskit(qc)
+
+    The ``my_qfunc`` function can now be used within QNodes, as a two-wire quantum
+    template. We can also pass ``wires`` when calling the returned template to define
+    which device wires it should operate on. If no wires are passed, it will default
+    to sequential wire labels starting at 0.
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit():
+            my_qfunc(wires=["a", "b"])
+            return qml.expval(qml.Z("a")), qml.var(qml.Z("b"))
+
+    >>> circuit()
+    (tensor(0.70738827, requires_grad=True),
+    tensor(0.99999937, requires_grad=True))
+
+    The measurements can also be passed directly to the function when creating the
+    qfunc, making it possible to create a PennyLane circuit with ``qml.QNode`:
+
+    >>> measurements = [qml.expval(qml.Z(0)), qml.var(qml.Z(1))]
+    >>> circuit = qml.QNode(qml.from_qiskit(qc, measurements), dev)
+    >>> circuit()
+    (tensor(0.70738827, requires_grad=True),
+    tensor(0.99999937, requires_grad=True))
+
+    .. note::
 
         The ``measurement`` keyword allows one to add a list of PennyLane measurements
-        that will override the terminal measurements present in their ``QuantumCircuit``.
+        that will **override** the terminal measurements present in the ``QuantumCircuit``,
```

Comment:
We say _terminal_ measurements here, but did we decide to only return terminal measurements? I thought we were also going to return the mid-circuit ones too?

Reply:
The way the converter is currently implemented, if no measurements are passed by the user, all of the mid-circuit measurements _and_ terminal measurements will be returned. If the user does specify `measurements` though, we only return the specified measurements (mid-circuit measurements will behave correctly but won't be returned).

Reply:
Thanks, I'm convinced it looks good now :pray: